experimental/dds/tree2/src/feature-libraries/sequence-field/format.ts

```diff
@@ -388,38 +322,58 @@ export const Detach = <Schema extends TSchema>(tNodeChange: Schema) =>
  * This mark should only exist as part of intermediate output of compose and should be removed during the amendCompose pass.
  */
 export interface MovePlaceholder<TNodeChange>
-	extends CellTargetingMark,
-		HasRevisionTag,
+	extends HasRevisionTag,
 		HasMoveId,
 		HasChanges<TNodeChange> {
 	type: "Placeholder";
-	count: NodeCount;
 }
 
-export interface Modify<TNodeChange = NodeChangeType> extends CellTargetingMark {
+export interface Modify<TNodeChange = NodeChangeType> {
 	type: "Modify";
 	changes: TNodeChange;
 }
 export const Modify = <Schema extends TSchema>(tNodeChange: Schema) =>
-	Type.Composite(
-		[
-			CellTargetingMark,
-			Type.Object({
-				type: Type.Literal("Modify"),
-				changes: tNodeChange,
-			}),
-		],
+	Type.Object(
+		{
+			type: Type.Literal("Modify"),
+			changes: tNodeChange,
+		},
 		noAdditionalProps,
 	);
 
-export type Mark<TNodeChange = NodeChangeType> =
-	| NoopMark
+export type Effect<TNodeChange = NodeChangeType> =
 	| Modify<TNodeChange>
 	| MovePlaceholder<TNodeChange>
 	| Attach<TNodeChange>
 	| Detach<TNodeChange>;
+export const Effect = <Schema extends TSchema>(tNodeChange: Schema) =>
+	Type.Union([Modify(tNodeChange), Attach(tNodeChange), Detach(tNodeChange)]);
+
+export interface Mark<TNodeChange> {
+	count: CellCount;
+	/**
+	 * Describes the detach which last emptied the target cells,
+	 * or the attach which allocated the cells if the cells have never been filled.
+	 * Undefined if the target cells are not empty in this mark's input context.
+	 */
+	cellId?: CellId;
```

Comment:
Does this rely on a scheme where when `cellId.revision` is undefined, we interpret it to be the revision of the mark?

Reply:
No, if `cellId.revision` is undefined (but `cellId` is defined), then the revision on the mark effect will also be undefined, and we interpret it to be the revision of the changeset.

On that note, `Insert.revision` can probably be removed (it should always be the same as `cellId.revision` right now), but I figured that doesn't matter much since it will be present again as part of the unification of insert, revive, move-in, and return-to into fill. Happy to remove it for now if you think that's worth it.